sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerProactiveInitConfig.java

```diff
@@ -0,0 +1,86 @@
+    private final List<CosmosContainerIdentity> cosmosContainerIdentities;
+    private final int numProactiveConnectionRegions;
+
+    CosmosContainerProactiveInitConfig(
+        List<CosmosContainerIdentity> cosmosContainerIdentities,
+        int numProactiveConnectionRegions) {
+
+        this.cosmosContainerIdentities = Collections.unmodifiableList(cosmosContainerIdentities);
+        this.numProactiveConnectionRegions = numProactiveConnectionRegions;
+    }
+
+    /**
+     * Gets the list of container identities. The returned list is protected against modifications.
+     *
+     * @return list of {@link CosmosContainerIdentity}
+     * */
+    public List<CosmosContainerIdentity> getCosmosContainerIdentities() {
+        return cosmosContainerIdentities;
+    }
+
+    /**
+     * Gets the no. of proactive connection regions
+     *
+     * <p>
+     * Proactive connection regions constitute those regions where replicas of container partitions have connections opened to prior
+     * to performing any workload on the container. This way the latency associated with opening connections
+     * does not impact the latency associated with performing workloads on the container. These connections are
+     * opened synchronously when the {@link CosmosClient}/{@link CosmosAsyncClient} is built.
+     * </p>
+     * <p>
+     * These proactive connection regions are a subset of the preferred regions configured through the {@link CosmosClientBuilder}. The first
+     * {@link CosmosContainerProactiveInitConfig#getNumProactiveConnectionRegions()} read regions from preferred regions are picked. In this context a write-region could also be a read-region but not vice-versa.
+     * </p>
+     * <p>
+     * Consider a multi-master account with client configured with preferred regions - "US West" (write-region) and "US East" (write-region)
+     * <br>
+     * 1. If the no. of proactive regions is set to two, connections to "US West" and "US East" are opened proactively.
+     * <br>
+     * 2. If the no. of proactive regions is set to one, connections to "US West" are opened proactively.
+     * <br><br>
+     * Consider a single-master account with client configured with preferred regions - "US West" (read-region), "US East" (read-region) and
+     * "West Europe" (write-region)
+     * <br>
+     * 1. If the no. of proactive regions is set to two, connections to "US West" and "US East" are opened proactively. If your application
+     * has workloads which are write-heavy it is important to prioritize write regions in the list of preferred regions.
+     * </p>
+     *
+     * @return no. of proactive connection regions
+     */
+    public int getNumProactiveConnectionRegions() {
+        return numProactiveConnectionRegions;
+    }
+
+    @Override
+    public String toString() {
+
+        if (this.cosmosContainerIdentities == null || cosmosContainerIdentities.isEmpty()) {
+            return "";
+        } else {
```

Comment:
nit: (and this is more of preference of code style) - you can remove the  else statement - just have the 2nd return without else block - it is technically the else case of the if statement since it did not go into 'return "";' - saves a bit of indentation and code lines and IMHO is more readable. But this is just a preference, can be left as it is.

Reply:
Agreed